scripts/support/clone-remote-database

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+set -euo pipefail
+start_tm=$(date '+%s')
+export TIME="%C %E"
+
+usage()
+{
+	set +xv
+	echo 1>&2 "$0 - clone a remote mysql database to a local machine for testing"
+	echo 1>&2 ""
+	echo 1>&2 "Usage: $0 [-l <LOCAL_DB>] [-r <REMOTE_DB>] REMOTE_HOST"
+	echo 1>&2 "REMOTE_HOST -- name or ip of remote host (passed to ssh)"
+	echo 1>&2 "LOCAL_DB -- name of new database to create locally (default: 'openboxes')"
+	echo 1>&2 "REMOTE_DB -- name of database to clone from REMOTE_HOST (default: 'openboxes')"
+	echo 1>&2 ""
+	echo 1>&2 "Use with care -- this script will destroy \$LOCAL_DB if it exists!!"
+	exit 1
+}
+
+#
+# Pretty-print the difference between two timestamps, $1 and $2.
+#
+elapsed_tm()
+{
+	set +e
+	# guess whether we're running GNU date: thanks for nothing, POSIX
+	date -j >/dev/null 2>&-
+	if [ $? -ne 0 ]
+	then
+		# GNU/linux
+		date -ud@$(($2 - $1)) "+%H:%M:%S"
+	else
+		# macOS/BSD
+		date -juf "%s" $(($2 - $1)) "+%H:%M:%S"
+	fi
+	set -e
+}
+
+LOCAL_DB="openboxes"
+REMOTE_DB="openboxes"
+while getopts "hl:r:" o
+do
+	case "${o}" in
+	l)
+		LOCAL_DB=${OPTARG}
+		[ "$LOCAL_DB" ] || usage
+		;;
+	r)
+		REMOTE_DB=${OPTARG}
+		[ "$REMOTE_DB" ] || usage
+		;;
+	h)
+		usage
+		;;
+	*)
+		usage
+		;;
+	esac
+done
+shift $((OPTIND-1))
+
+[ "$1" ] && [ "$#" -eq 1 ] || usage
+REMOTE_HOST="$1"
+
+
+
+SQL_BASENAME="$REMOTE_HOST-$REMOTE_DB"
+
+echo "Clobbering local database $LOCAL_DB ..."
+mysql -e "drop database if exists \`$LOCAL_DB\`; create database \`$LOCAL_DB\` default charset utf8; grant all on \`$LOCAL_DB\`.* to 'openboxes'@'localhost' identified by 'openboxes';"
+mysql $LOCAL_DB -e 'select 1' > /dev/null
+
+# prevent "ERROR 2006 (HY000): MySQL server has gone away"
+# see also https://stackoverflow.com/questions/10474922/error-2006-hy000-mysql-server-has-gone-away
+mysql -e "set global max_allowed_packet=1024*1024*1024;"
+
+echo -n "Counting products in remote database $REMOTE_HOST:$REMOTE_DB ..."
+remote_product_cnt=$(ssh "$REMOTE_HOST" "mysql $REMOTE_DB -Nse 'select count(*) from product;'")
```

Comment:
It probably doesn't matter but since we're potentially touching a production database, I'd like to reduce our footprint as much as possible.
```
select count(id) from product
```